tarantool/dbapi.py

```diff
@@ -0,0 +1,307 @@
+
+        The cursor will be unusable from this point forward; an Error (or
+        subclass) exception will be raised if any operation is attempted with
+        the cursor.
+        """
+        self._c = None
+
+    @staticmethod
+    def _convert_param(p):
+        if p is None:
+            return "NULL"
+        if isinstance(p, bool):
+            return str(p)
+        if isinstance(p, str):
+            return "'%s'" % p.replace("'", "''")
+        return "'%s'" % p
+
+    @staticmethod
+    def _extract_last_row_id(body):
+        try:
+            val = next(iter(body.values())).get(1)[-1]
+        except TypeError:
+            val = -1
+        return val
+
+    def execute(self, query, params=None):
+        """
+        Prepare and execute a database operation (query or command).
+
+        Parameters may be provided as sequence or mapping and will be bound
+        to variables in the operation. Variables are specified in a
+        database-specific notation (see the module's paramstyle attribute for
+        details).
+
+        A reference to the operation will be retained by the cursor. If the
+        same operation object is passed in again, then the cursor can
+        optimize its behavior. This is most effective for algorithms where
+        the same operation is used, but different parameters are bound to it
+        (many times).
+
+        For maximum efficiency when reusing an operation, it is best to use
+        the .setinputsizes() method to specify the parameter types and sizes
+        ahead of time. It is legal for a parameter to not match the
+        predefined information; the implementation should compensate,
+        possibly with a loss of efficiency.
+
+        The parameters may also be specified as list of tuples to e.g. insert
+        multiple rows in a single operation, but this kind of usage is
+        deprecated: .executemany() should be used instead.
+
+        Return values are not defined.
+        """
+        if self.closed:
+            raise self._c.ProgrammingError
+
+        if params:
+            query = query % tuple(
+                self._convert_param(param) for param in params)
+
+        response = self._c.execute(query)
+        self.rows = tuple(response.data) if len(
+            response.body) > 1 else None
+
+        if update_insert_pattern.match(query):
+            try:
+                self._rowcount = response.rowcount
+            except InterfaceError:
+                self._rowcount = 1
+        else:
+            self._rowcount = -1
+
+        if query.upper().startswith('INSERT'):
+            self._lastrowid = self._extract_last_row_id(response.body)
+        return response
+
+    def executemany(self, query, param_sets):
+        rowcounts = []
+        for params in param_sets:
+            self.execute(query, params)
+            rowcounts.append(self.rowcount)
+
+        self._rowcount = -1 if -1 in rowcounts else sum(rowcounts)
+        return self
```

Comment:
If we'll not return anything, a caller will not use return value anyhow. So we'll be able to return some useful value from this method in a future implementation without breaking backward compatibility.

So I agree, it would be better to don't return anything from `execute*()` methods in the first implementation.